### ISSUE TITLE
Make map generator a generator

### ISSUE THREAD

**Issue by NotAFile** [piqueserver/piqueserver#131]
The map generator is currently just an object that ends up reimplementing the generator API poorly, it would be nice to switch it to an actual generator instead.

**Comment by davidmalmstrom** [comment]
Is this referring to the ProgressiveMapGenerator object [here](https://github.com/piqueserver/piqueserver/blob/49317cbe23a9b479f11193de8e2575fea92b5030/pyspades/mapgenerator.py) or the C++ object starting [here](https://github.com/piqueserver/piqueserver/blob/49317cbe23a9b479f11193de8e2575fea92b5030/pyspades/vxl_c.cpp#L408)? Or both?

**Comment by NotAFile** [comment]
Yes, both. But if you had to only do one, making the non-cython object a generator would be better.
